project/src/components/offer-card/offer-card.tsx

```diff
@@ -1,30 +1,50 @@
-import { Link } from 'react-router-dom';
+import { Link, generatePath } from 'react-router-dom';
+import cn from 'classnames';
+
+import { AppRoute } from '../../const';
+
+import { Offer } from '../../types/offer';
 
 type PropsType = {
-  id?: number;
-  price: number;
-  isPremium: boolean;
-  isFavorite: boolean;
-  imgPath: string;
-  title: string;
-  roomType: string;
+  offer: Offer;
+  onOfferCardMouseEnter: (id: number) => void;
+  activeCard: number;
+  page: string;
 }
 
-function OfferCard({id, price, isPremium, isFavorite, imgPath, title, roomType}: PropsType): JSX.Element {
+function OfferCard({offer, page, onOfferCardMouseEnter, activeCard}: PropsType): JSX.Element {
+  const {id, price, isPremium, isFavorite, imgPath, title, roomType} = offer;
   const favoriteClassName = 'place-card__bookmark-button button';
-  const offerLink = `offer/${id}`;
+
+  const handleMouseEnter = () => {
+    onOfferCardMouseEnter(id);
+  };
 
   return (
-    <>
+    <article key={id} className={cn('place-card',
+      {
+        'cities__place-card': page === 'main',
+        'favorites__card': page === 'favorites',
+      })} onMouseEnter={handleMouseEnter}
```

Comment:
Отдельно можно не выносить 
```suggestion
      })} onMouseEnter={() => onOfferCardMouseEnter(id)}
```